pyiron_base/jobs/job/extension/executable.py

```diff
@@ -299,7 +299,10 @@ def _executable_select(self):
         try:
             return self.executable_lst[self.version]
         except KeyError:
-            return ""
+            if isinstance(self.version, str):
+                return self.version
+            else:
+                return ""
```

Comment:
I am slightly confused - why does the `version` hold the `str: absolute executable path` in this case?

Reply:
In the current implementation when the `executable_path` is set this internally is stored in the `version` property: 
https://github.com/pyiron/pyiron_base/blob/main/pyiron_base/jobs/job/extension/executable.py#L205

I added the comparison of the storage dictionary to demonstrate this. I am also open to changing the executable interface but from my perspective this goes beyond the scope of this pull request.